### ISSUE TITLE
Error on calling climate.set_preset_mode from an automation

### ISSUE THREAD

**Issue by ejalal** [ScratMan/HASmartThermostat#28]
Hello

I just replaced one of my generic_thermostat(s) with this smart_thermostat to experiment with and see if it could be a long term replacement for all my other thermostats, and I'm having an error fitting it with my existing automations (with node red)

I have this error when I call climate.set_preset_mode (inherited from the old generic_thermostat), any help would be much appreciated.

Testing from HA Developer Tools gives the same result

```
service: climate.set_preset_mode
data:
  preset_mode: none
target:
  entity_id:
    - climate.bathroom

```
```
2021-11-29 08:57:04 ERROR (SyncWorker_17) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 863, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 704, in _handle_entity_call
    await result
  File "/config/custom_components/smart_thermostat/climate.py", line 649, in async_set_preset_mode
    self._target_temp = self.presets[preset_mode]
KeyError: 'none'

```

**Comment by ScratMan** [comment]
Hi, issue confirmed. It occurs when trying to switch preset mode from 'none' to 'none'.
A commit will be added to next release.

**Comment by ScratMan** [comment]
Closed with [release v2021.11.9](https://github.com/ScratMan/HASmartThermostat/releases/tag/2021.11.9)
